packages/alchemy/src/react/hooks/useAddPasskey.ts

```diff
@@ -0,0 +1,45 @@
+"use client";
+
+import { useMutation, type UseMutateFunction } from "@tanstack/react-query";
+import { ClientOnlyPropertyError } from "../../config/errors.js";
+import { useAlchemyAccountContext } from "../context.js";
+import { useSigner } from "./useSigner.js";
+
+export type UseAddPasskeyResult = {
+  addPasskey: UseMutateFunction<
+    string[],
+    Error,
+    CredentialCreationOptions | undefined | void,
+    unknown
+  >;
+  isAddingPasskey: boolean;
+  error: Error | null;
+};
+
+export function useAddPasskey(): UseAddPasskeyResult {
+  const { queryClient } = useAlchemyAccountContext();
+  const signer = useSigner();
+
+  const {
+    mutate: addPasskey,
+    isPending: isAddingPasskey,
+    error,
+  } = useMutation(
+    {
+      mutationFn: async (params?: CredentialCreationOptions | void) => {
```

Comment:
reason I did `... | void`: without it, react query makes you explicit do `undefined` for optional inputs to a mutation.

ref: https://github.com/TanStack/query/issues/1077#issuecomment-1698583282.